web_src/js/index.js

```diff
@@ -357,214 +437,221 @@ function initSimpleMDEImagePaste(simplemde, files) {
 
-      const listIds = [];
-      $(this).parent().find('.item').each(function () {
-        if ($(this).hasClass('checked')) {
-          listIds.push($(this).data('id'));
-          $($(this).data('id-selector')).removeClass('hide');
+      updateIssuesMeta(
+        $listMenu.data('update-url'),
+        '',
+        $listMenu.data('issue-id'),
+        $(this).data('id'),
+        $(this).data('is-checked')
+      );
+      $listMenu.data('action', 'update'); // Update to reload the page when we updated items
+      return false;
+    }
+
+    if ($(this).hasClass('checked')) {
+      $(this).removeClass('checked');
+      $(this).find('.octicon-check').addClass('invisible');
+      if (hasLabelUpdateAction) {
+        if (!($(this).data('id') in labels)) {
+          labels[$(this).data('id')] = {
+            'update-url': $listMenu.data('update-url'),
+            action: 'detach',
+            'issue-id': $listMenu.data('issue-id'),
+          };
         } else {
-          $($(this).data('id-selector')).addClass('hide');
+          delete labels[$(this).data('id')];
         }
-      });
-      if (listIds.length === 0) {
-        $noSelect.removeClass('hide');
+      }
+    } else {
+      $(this).addClass('checked');
+      $(this).find('.octicon-check').removeClass('invisible');
+      if (hasLabelUpdateAction) {
+        if (!($(this).data('id') in labels)) {
+          labels[$(this).data('id')] = {
+            'update-url': $listMenu.data('update-url'),
+            action: 'attach',
+            'issue-id': $listMenu.data('issue-id'),
+          };
+        } else {
+          delete labels[$(this).data('id')];
+        }
+      }
+    }
+
+    const listIds = [];
+    $(this).parent().find('.item').each(function () {
+      if ($(this).hasClass('checked')) {
+        listIds.push($(this).data('id'));
+        $($(this).data('id-selector')).removeClass('hide');
       } else {
-        $noSelect.addClass('hide');
+        $($(this).data('id-selector')).addClass('hide');
       }
-      $($(this).parent().data('id')).val(listIds.join(','));
-      return false;
     });
-    $listMenu.find('.no-select.item').on('click', function () {
-      if (hasLabelUpdateAction || selector === 'select-assignees-modify') {
-        updateIssuesMeta(
-          $listMenu.data('update-url'),
-          'clear',
-          $listMenu.data('issue-id'),
-          '',
-          ''
-        ).then(reload);
-      }
-
-      $(this).parent().find('.item').each(function () {
-        $(this).removeClass('checked');
-        $(this).find('.octicon').addClass('invisible');
-        $(this).data('is-checked', 'remove');
-      });
-
-      $list.find('.item').each(function () {
-        $(this).addClass('hide');
-      });
+    if (listIds.length === 0) {
       $noSelect.removeClass('hide');
-      $($(this).parent().data('id')).val('');
+    } else {
+      $noSelect.addClass('hide');
+    }
+    $($(this).parent().data('id')).val(listIds.join(','));
```

Comment:
Better use `.closest(selector)` instead of `.parent` to not depend so tightly on DOM structure.